### PR TITLE
adguardhome: enable legacy openssl provider for nodejs and bump to 0.107.25

### PR DESCRIPTION
**Pull request by dobo90**
Maintainer: me
Compile tested: aarch64_generic, NanoPi R2S, OpenWrt SNAPSHOT r22247
Run tested: aarch64_generic, NanoPi R2S, OpenWrt SNAPSHOT r22247

Description:
**adguardhome: enable legacy openssl provider for nodejs**

Currently compilation fails because of:
```
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
```

What's interesting package gets built but when trying to access UI there's
`404: page not found` error.

It has been reported in multiple places:
* https://github.com/AdguardTeam/AdGuardHome/issues/5559
* https://github.com/AdguardTeam/AdGuardHome/issues/4595

**adguardhome: bump to 0.107.25**

* Full changelog available at:
  * https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.25

### CODE REVIEW

File: net/adguardhome/Makefile [hunk 1]
```diff
@@ -56,7 +56,7 @@ endef
 define Build/Compile
 	( \
 		pushd $(PKG_BUILD_DIR) ; \
-		make js-deps js-build ; \
+		NODE_OPTIONS=--openssl-legacy-provider make js-deps js-build ; \
```
Comment (PussAzuki) [inline net/adguardhome/Makefile:59]: `export NODE_OPTIONS=--openssl-legacy-provider ; \`
`make js-deps js-build ; \`

I see they recommend this writing style, and it works.

Comment (1715173329) [reply net/adguardhome/Makefile:59]: There's no difference in this case.

Comment (PussAzuki) [reply net/adguardhome/Makefile:59]: Just to make it look more comfortable...

Comment (1715173329) [reply net/adguardhome/Makefile:59]: It's just a personal preference. tbh I don't like making local variables global.

**Comment by peterwillcn** [comment]
404

**Pull request merged**